src/DependencyRewriterV1.php

```diff
@@ -0,0 +1,168 @@
+    {
+        $this->output(sprintf('<info>In %s</info>', __METHOD__), IOInterface::DEBUG);
+        $operation = $event->getOperation();
+
+        switch (true) {
+            case $operation instanceof Operation\InstallOperation:
+                $package = $operation->getPackage();
+                break;
+            case $operation instanceof Operation\UpdateOperation:
+                $package = $operation->getTargetPackage();
+                break;
+            default:
+                // Nothing to do
+                $this->output(sprintf(
+                    '<info>Exiting; operation of type %s not supported</info>',
+                    get_class($operation)
+                ), IOInterface::DEBUG);
+                return;
+        }
+
+        $name = $package->getName();
+        if (! $this->isZendPackage($name)) {
+            // Nothing to do
+            $this->output(sprintf(
+                '<info>Exiting; package "%s" does not have a replacement</info>',
+                $name
+            ), IOInterface::DEBUG);
+            return;
+        }
+
+        $replacementName = $this->transformPackageName($name);
+        if ($replacementName === $name) {
+            // Nothing to do
+            $this->output(sprintf(
+                '<info>Exiting; while package "%s" is a ZF package, it does not have a replacement</info>',
+                $name
+            ), IOInterface::DEBUG);
+            return;
+        }
+
+        $version = $package->getVersion();
+        $replacementPackage = $this->composer->getRepositoryManager()->findPackage($replacementName, $version);
+
+        if ($replacementPackage === null) {
+            // No matching replacement package found
+            $this->output(sprintf(
+                '<info>Exiting; no replacement package found for package "%s" with version %s</info>',
+                $replacementName,
+                $version
+            ), IOInterface::DEBUG);
+            return;
+        }
+
+        $this->output(sprintf(
+            '<info>Replacing package %s with package %s, using version %s</info>',
+            $name,
+            $replacementName,
+            $version
+        ), IOInterface::VERBOSE);
+
+        $this->replacePackageInOperation($replacementPackage, $operation);
+    }
+
+    private function replacePackageInOperation(PackageInterface $replacement, Operation\OperationInterface $operation)
+    {
+        $this->updateProperty(
+            $operation,
+            $operation instanceof Operation\UpdateOperation ? 'targetPackage' : 'package',
+            $replacement
+        );
+    }
+
+    /**
+     * @param object $object
+     * @param string $property
+     * @param mixed $value
+     */
+    private function updateProperty($object, $property, $value)
+    {
+        $r = new ReflectionProperty($object, $property);
+        $r->setAccessible(true);
+        $r->setValue($object, $value);
```

Comment:
can use closure bind, something like:

```php
    (function ($object, $property, $value) {
        $object->$property = $value;
    })->bindTo($object, $object)($object, $property, $value);
```

ref https://ocramius.github.io/blog/accessing-private-php-class-members-without-reflection/

Reply:
This is all copy & pasted from current version.
I know the closure thing but didnt want to change current behavior.